src/io/utilities/readSBML.m

```diff
@@ -636,10 +640,10 @@
     model.lb = fbc_lb;
     model.ub = fbc_ub;
     if noObjective==0; % when there is an objective function
-        indexObj=findRxnIDs(model,fbc_obj);
+        indexObj=findRxnIDs(model,cleanUpFormatting(fbc_obj));
         % indexObj=find(strcmp(fbc_obj,model.rxns))
-        model.c(indexObj)=1;
-        model.osense = - sign(fbc_obj_value);
+        model.c(indexObj) = fbc_obj_value(:);
+        model.osense = fbc_obj_sense;
```

Comment:
@shjchan,
I think it would be best if you add the test code to the existing testReadSBML (test/verifiedTest/testSBML/testReadSBML.m). but yes, this should be sufficient.